test/test_support.ts

```diff
@@ -159,113 +172,64 @@ export function getLineAndColumn(source: string, token: string): {line: number,
-  useTransformer: boolean;
-  /**
-   * Controls the name of the file produced by the compiler.  If there's more
-   * than one input file, they'll all be concatenated together in the outFile
-   */
-  outFile?: string;
-  filesNotToProcess: Set<string>;
-  inlineSourceMap: boolean;
-  generateDTS: boolean;
+export function assertSourceMapping(
+    compiledJS: string, sourceMap: SourceMapConsumer, element: string,
+    expectedPosition: Partial<{line: number, column: number, source: string}>) {
+  const {line, column} = getLineAndColumn(compiledJS, element);
+  const originalPosition = sourceMap.originalPositionFor({line, column});
+  if (expectedPosition.line) {
+    expect(originalPosition.line).to.equal(expectedPosition.line);
+  }
+  if (expectedPosition.column) {
+    expect(originalPosition.column).to.equal(expectedPosition.column);
+  }
+  if (expectedPosition.source) {
+    expect(originalPosition.source).to.equal(expectedPosition.source);
+  }
 }
 
-const DEFAULT_COMPILER_OPTIONS = {
-  filesNotToProcess: new Set<string>(),
-  inlineSourceMap: false,
-  generateDTS: false,
+export const sourceMapCompilerOptions: ts.CompilerOptions = {
+  inlineSources: true,
+  experimentalDecorators: true,
+  declaration: true,
+  // Disable searching for @types typings. This prevents TS from looking
+  // around for a node_modules directory.
+  types: [],
+  target: ts.ScriptTarget.ES2015,
+  skipDefaultLibCheck: true,
+  module: ts.ModuleKind.CommonJS,
+  sourceMap: true,
 };
 
-export function compile(
-    sources: Map<string, string>,
-    partialOptions: Partial<CompilerOptions>&{useTransformer: boolean}): {
-  compiledJS: string,
-  dts: string | undefined,
-  sourceMap: SourceMapConsumer,
-  sourceMapText: string
-} {
-  const options: CompilerOptions = {...DEFAULT_COMPILER_OPTIONS, ...partialOptions};
-  const resolvedSources = new Map<string, string>();
-  for (const fileName of toArray(sources.keys())) {
-    resolvedSources.set(ts.sys.resolvePath(fileName), sources.get(fileName)!);
-  }
-
-  let compilerOptions: ts.CompilerOptions;
-  if (options.inlineSourceMap) {
-    compilerOptions = {
-      inlineSourceMap: options.inlineSourceMap,
-      inlineSources: true,
-      outFile: options.outFile,
-      experimentalDecorators: true,
-      declaration: options.generateDTS,
-    };
-  } else {
-    compilerOptions = {
-      sourceMap: true,
-      inlineSources: true,
-      outFile: options.outFile,
-      experimentalDecorators: true,
-      declaration: options.generateDTS,
-    };
-  }
+/**
+ * Controls the name of the file produced by the compiler.  If there's more
+ * than one input file, they'll all be concatenated together in the outFile
+ */
+export function generateOutfileCompilerOptions(outFile: string): ts.CompilerOptions {
+  return {
+    outFile,
+    module: ts.ModuleKind.None,
+  };
+}
 
-  const fileNames = toArray(sources.keys());
+export const inlineSourceMapCompilerOptions: ts.CompilerOptions = {
```

Comment:
It's used in e2e_source_map_tests, which was collapsed by the github UI as having too many changes to one file. You should go review it!